src/database/adatabasesetup.cpp

```diff
@@ -292,6 +286,11 @@ bool ADataBaseSetup::downloadTemplates()
         dl->setFileName(template_dir.filePath(table % QStringLiteral(".csv")));
         dl->download();
         loop.exec(); // event loop waits for download done signal before allowing loop to continue
+
+        QFileInfo downloaded_file(template_dir.filePath(table % QStringLiteral(".csv")));
+        if (downloaded_file.size() == 0)
+            return false; // ssl/network error
+
```

Comment:
Are we sure there is no leak here?

Reply:
Yes, since they are created before the main event loop is started and marked for deletion with deleteLater(), they are deleted at this point, see [here](https://doc.qt.io/qt-5/qobject.html#deleteLater).